src/main/java/edu/tamu/sage/controller/DiscoveryViewController.java

```diff
@@ -104,8 +105,12 @@ public ApiResponse findBySlug(@PathVariable String slug, @RequestParam(name = "f
                 PageRequest pageRequest = PageRequest.of(page.getPageNumber(), page.getPageSize(),Sort.by(defaultDirection, field));
                 page = pageRequest;
             } else {
-                String sort = page.getSort().toString().split(":")[0];
-                page = PageRequest.of(page.getPageNumber(), page.getPageSize(),Sort.Direction.fromString(direction), sort);
+                String sort = (page.getSort().toString().split(":")[0] != "UNSORTED") ? page.getSort().toString().split(":")[0] : discoveryView.getResourceLocationUriKey().replaceAll("[{}]", "");
```

Comment:
String comparison should use equals method. The split operation only needs to happen once.  The default direction of the discovery view should be respected. Some additional suggested cleanup.

https://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/EnumUtils.html#isValidEnum-java.lang.Class-java.lang.String-

```
Direction defaultDirection = discoveryView.isAscending() ? Direction.ASC : Direction.DESC;
...

String sort = page.getSort().toString().split(":")[0];
sort = !sort.equals("UNSORTED") ? sort : discoveryView.getResourceLocationUriKey().replaceAll("[{}]", "");
Direction dir = EnumUtils.isValidEnum(Direction.class, direction) ? Direction.valueOf(direction) : defaultDirection ;
page = PageRequest.of(page.getPageNumber(), page.getPageSize(), dir, sort);
```

Reply:
Utilized this block